### ISSUE TITLE
[Accessibility] Make tags accessible

### ISSUE THREAD

**Issue by andresgalante** [auth0/cosmos#1295]
The Tags is not accessible, what needs to happen to make it accessible?

- The `a` should be a `button`
- Make sure the `button` has focus
- when used in a group add `aria-labelledby="[id value of .pf-c-button]"` to gives the button an accessible name by referring to the element that provides the position of the button within a list. Required when the button is being removed. 
- Make sure touch areas are accesible
